src/signals/incident-management/containers/DefaultTextsAdmin/components/DefaultTextsForm/index.js

```diff
@@ -21,232 +29,244 @@ const StyledButton = styled(Button)`
-    this.props.onOrderDefaultTexts({ index, type });
-    this.form.updateValueAndValidity();
-  }
+    onOrderDefaultTexts({ index, type });
+    form.updateValueAndValidity();
+  };
 
-  render() {
-    return (
-      <div className="default-texts-form container">
-        <FieldGroup
-          control={this.form}
-          render={({ invalid }) => (
-            <form
-              onSubmit={this.handleSubmit}
-              className="default-texts-form__form"
-            >
-              <FieldControlWrapper
-                render={HiddenInput}
-                name="state"
-                control={this.form.get('state')}
-              />
-
-              <FieldControlWrapper
-                render={HiddenInput}
-                name="categoryUrl"
-                control={this.form.get('categoryUrl')}
-              />
-
-              {this.items.map((item, index) => (
-                <div key={item} className="row default-texts-form__row">
-                  <div className="col-8">
-                    <FieldControlWrapper
-                      placeholder="Titel"
-                      render={TextInput}
-                      name={`title${index}`}
-                      control={this.form.get(`${item}.title`)}
-                    />
-
-                    <FieldControlWrapper
-                      placeholder="Tekst"
-                      render={TextAreaInput}
-                      name={`text${index}`}
-                      control={this.form.get(`${item}.text`)}
-                    />
-                  </div>
-                  <div className="col-2 default-texts-form__actions">
-                    <StyledButton
-                      size={44}
-                      variant="blank"
-                      disabled={
-                        index === 0 || !this.form.get(`${item}.text`).value
-                      }
-                      iconSize={16}
-                      icon={<ChevronUp />}
-                      onClick={e => this.changeOrdering(e, index, 'up')}
-                    />
-                    <StyledButton
-                      size={44}
-                      variant="blank"
-                      disabled={
-                        index === this.items.length - 1
-                        || !this.form.get(`item${index + 1}.text`).value
-                      }
-                      iconSize={16}
-                      icon={<ChevronDown />}
-                      onClick={e => this.changeOrdering(e, index, 'down')}
-                    />
-                  </div>
-                </div>
-              ))}
-
-              <Button variant="secondary" type="submit" disabled={invalid}>
-                Opslaan
-              </Button>
-            </form>
-          )}
-        />
-      </div>
-    );
-  }
-}
+  const subscribe = () => {
+    items.forEach((item, index) => {
+      form.get(item).valueChanges.subscribe(data => {
+        onSaveDefaultTextsItem({ index, data });
```

Comment:
Why is there both a `handleSubmit` handler and the handler on this line? Don't they do the same?

Reply:
that is a good point we can remove that

Reply:
I removed all related parts